fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -435,6 +435,9 @@ private void init(String s, Type type) throws AnalysisException {
                 this.roundFloor(((ScalarType) type).getScalarScale());
             }
             this.type = type;
+            if (checkRange() || checkDate()) {
```

Comment:
Should it be:
```
if (!checkRange() || !checkDate())
```
?

Reply:
`checkRange` and `checkDate` return true is error 😂
https://github.com/yangzhg/doris/blob/881670566c0aa577dd83af69c7b3d9f7a3986ab2/fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java#L1298